src/sage/categories/examples/magmas.py

```diff
@@ -73,7 +72,7 @@ def __init__(self, alphabet=('a', 'b', 'c', 'd')):
             raise ValueError("alphabet must not contain characters "
                              "'(', ')' or '*'")
         self.alphabet = alphabet
-        Parent.__init__(self, category=Magmas().FinitelyGenerated())
+        Parent.__init__(self, category=Magmas().FinitelyGenerated().Infinite())
```

Comment:
well, unless the set of generators is empty..

Reply:
I'll just wait until the bots are done, then I'd apply
```suggestion
        cat = Magmas().FinitelyGenerated()
        if alphabet:
            Parent.__init__(self, category=cat.Infinite())
        else:
            Parent.__init__(self, category=cat)
```

Reply:
Actually, it might be better to require that there is at least one generator.  Otherwise, it would be necessary to somehow state that the magma is empty.